bundle/deploy/terraform/pkg.go

```diff
@@ -19,21 +22,54 @@ const (
 	TerraformProviderVersionEnv = "DATABRICKS_TF_PROVIDER_VERSION"
 )
 
+// TerraformVersion represents the version of the Terraform CLI to use.
+// It allows for users overriding the default version.
+type TerraformVersion struct {
+	Version *version.Version
+
+	// These hashes are not used inside the CLI. They are only co-located here to be
+	// output in the "databricks bundle debug terraform" output. Downstream applications
+	// like the CLI docker image use these checksums to verify the integrity of the
+	// downloaded Terraform archive.
+	ChecksumLinuxArm64 string
+	ChecksumLinuxAmd64 string
+}
+
 // Terraform CLI version to use and the corresponding checksums for it. The
 // checksums are used to verify the integrity of the downloaded binary. Please
 // update the checksums when the Terraform version is updated. The checksums
 // were obtained from https://releases.hashicorp.com/terraform/1.5.5.
-//
-// These hashes are not used inside the CLI. They are only co-located here to be
-// output in the "databricks bundle debug terraform" output. Downstream applications
-// like the CLI docker image use these checksums to verify the integrity of the
-// downloaded Terraform archive.
-var TerraformVersion = version.Must(version.NewVersion("1.5.5"))
+var defaultTerraformVersion = TerraformVersion{
+	Version: version.Must(version.NewVersion("1.5.5")),
 
-const (
-	checksumLinuxArm64 = "b055aefe343d0b710d8a7afd31aeb702b37bbf4493bb9385a709991e48dfbcd2"
-	checksumLinuxAmd64 = "ad0c696c870c8525357b5127680cd79c0bdf58179af9acd091d43b1d6482da4a"
-)
+	ChecksumLinuxArm64: "b055aefe343d0b710d8a7afd31aeb702b37bbf4493bb9385a709991e48dfbcd2",
+	ChecksumLinuxAmd64: "ad0c696c870c8525357b5127680cd79c0bdf58179af9acd091d43b1d6482da4a",
+}
+
+// GetTerraformVersion returns the Terraform version to use.
+// The user can configure the Terraform version to use by setting the
+// DATABRICKS_TF_VERSION environment variable to the desired version.
+// It returns an error if the version is malformed.
+func GetTerraformVersion(ctx context.Context) (TerraformVersion, error) {
+	versionEnv, ok := env.Lookup(ctx, TerraformVersionEnv)
+	if !ok {
+		return defaultTerraformVersion, nil
+	}
+
+	v, err := version.NewVersion(versionEnv)
+	if err != nil {
+		return TerraformVersion{}, err
+	}
+
+	return TerraformVersion{
+		Version: v,
+
+		// Checksums are unknown if we return a user-specified version. This is not an issue,
+		// because they are only used in the output of the `databricks bundle debug terraform` command.
```

Comment:
Does it mean that the checksums are not actually validated?

Reply:
They are in the `hc-install` library, outside our purview.

These checksums are here to print in the `databricks bundle debug terraform` command.